File: deck-service/src/main/java/com/learnwords/deckservice/dto/deck/SentenceGenerationRequestDto.java
package com.learnwords.deckservice.dto.deck;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.learnwords.common.dto.WordItemDto;
import lombok.Builder;

import java.util.List;

@Builder
public record SentenceGenerationRequestDto(
    @JsonProperty("id")
    String id,
    @JsonProperty("requested_by_user_id")
    String requestedByUserId,
    @JsonProperty("words")
    List<WordItemDto> words,
    @JsonProperty("level")
    String level,
    @JsonProperty("category")
    String category,
    @JsonProperty("language_from")
    String languageFrom,
    @JsonProperty("language_to")
    String languageTo
) {
}

File: koog-service/src/main/kotlin/com/learnwords/koogservice/messaging/dto/SentenceGenerationRequestDto.kt
package com.learnwords.koogservice.messaging.dto

/**
 * DTO żądania generowania zdań z Kafki.
 *
 * @param requestedByUserId ID użytkownika, który zlecił generowanie
 * @param words lista słówek do wygenerowania zdań
 * @param level poziom językowy (A1, A2, B1, B2, C1, C2)
 * @param category kategoria tematyczna
 */
data class SentenceGenerationRequestDto(
    val id: String,
    val requestedByUserId: String,
    val words: List<WordItemDto>,
    val level: String = "B1",
    val category: String = "general",
    val languageFrom: String,
    val languageTo: String,
)


